src/api/data/cache/cache.go

```diff
@@ -127,18 +127,18 @@ func (c *cachedCharts) Search(params operations.SearchChartsParams) ([]*models.C
 // Refresh is the interface implementation for data.Charts
 // It refreshes cached data for all authoritative repository+chart data
 func (c *cachedCharts) Refresh() error {
-	c.rwm.Lock()
```

Comment:
I am not trying to optimize for running `Refresh()` multiple times, actually in theory there should be only one goRoutine being executed at once (unless it takes longer than the refresh period).

What I want to allow is that during the refresh process we keep accepting API calls. 

With the current implementation, assuming that for example the full refresh took 20 seconds (it downloads and process stuff). That would mean that no API requests will be accepted because of the Lock in Refresh and Rlocks in the auxiliare cache methods during all that time. https://github.com/helm/monocular/blob/master/src/api/data/cache/cache.go#L85

With the new implementation:

Downloads of new charts and assets happen without blocking the structure, just locking it at the moment of the content swap, having virtually no downtime.

> The only issue I see with concurrency here is that because all refresh calls share the same pool of workers, it would be possible to negatively impact the performance of the refresh merely by calling the function a whole bunch of times.

Could you elaborate? Is this related to tweaking the number of workers? Do you have any suggestions on how to measure and tweak it?